## Covariates File Format
```
One can provide a covariates file if needed for fitting LMM if necessary. GEMMA fits a linear
mixed model with an intercept term if no covariates file is provided, but does not internally provide
an intercept term if a covariates file is available. Therefore, if one has covariates other than the
intercept and wants to adjust for those covariates (W) simultaneously, one should provide GEMMA
with a covariates file containing an intercept term explicitly. The covariates file is similar to the
above BIMBAM multiple phenotype file, and must contain a column of 1â€™s if one wants to include
an intercept. An example covariates file with five individuals and three covariates (the first column
is the intercept) is as follows:
1 1 -1.5
1 2 0.3
1 2 0.6
1 1 -0.8
1 1 2.0
It can happen, especially in a small GWAS data set, that some of the covariates will be identical
to some of the genotypes (up to a scaling factor). This can cause problems in the optimization
algorithm and evoke GSL errors. To avoid this, one can either regress the phenotypes on the
covariates and use the residuals as new phenotypes, or use only SNPs that are not identical to any
of the covariates for the analysis. The later can be achieved, for example, by performing a standard
linear regression in the genotype data, but with covariates as phenotypes.
```
## Covariate file

Add 1s in first column and phenotype on second column

[Red phenotype as covariate](otherFiles/red-cov.txt)

[Yellow phenotype as covariate](otherFiles/red-cov.txt)

## Run covariate included mixed model 

```ruby
module load gemma

yellow=filtered/yellow
venation=filtered/venation
class=filtered/class
red=filtered/red

related_red=output/std_rel_red.sXX.txt
related_yellow=output/std_rel_yellow.sXX.txt


#gemma -bfile ${red} -k ${related_red} -c yellow-cov.txt -lmm 4 -o red_yelCov
gemma -bfile ${yellow} -k ${related_red} -c red-cov.txt -lmm 4 -o yellow_redCov
```

Output: 
